app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java

```diff
@@ -10,13 +11,17 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import fr.neamar.kiss.loader.LoadSearchPojos;
 import fr.neamar.kiss.pojo.Pojo;
 import fr.neamar.kiss.pojo.SearchPojo;
 
 public class SearchProvider extends Provider<SearchPojo> {
     private SharedPreferences prefs;
+    public static final String URL_REGEX = "^((https?|ftp)://|(www|ftp)\\.)?[a-z0-9-]+(\\.[a-z0-9-]+)+([/?].*)?$";
```

Comment:
Please add `?:` after all opening parenthesis (`(`), ie:
`"^(?:(?:https?|ftp)://|(?:www|ftp)[.])?[a-z0-9-]+(?:[.][a-z0-9-]+)+(?:[/].*)?$"`

Also you regexp doesn't handle [IDNA](https://en.wikipedia.org/wiki/Internationalized_domain_name)s correctly (non-ASCII extension for DNS) and the `(?:www|ftp)[.]` is redundant. Also hard-coding `http`, `https` & `ftp` isn't really a problem, but very unnecessary. May I therefor propose the following regexp that also handles the mentioned edge-cases:

`"^(?:[a-z]+://)?(?:[a-z0-9-]|[^\x00-\x7F])+(?:[.](?:[a-z0-9-]|[^\x00-\x7F])+)+.*$"`

For instance these things work (that previously didn't):

 * `http://evas-nähwerkstatt.at/`(Yes, that URL is actually advertised with an `ä`, hence this complaint)
 * `www.blab.com/` (Still works)
 * `gopher://wtf.invalid/` (if you had a [gopher](https://en.wikipedia.org/wiki/Gopher_(protocol)) client :wink:)